src/Metadata/Property/PropertyMetadata.php

```diff
@@ -60,7 +60,7 @@ public function getType()
      *
      * @param Type $type
      *
-     * @return self
+     * @return PropertyMetadata
```

Comment:
But as you said it's still a draft, it might change. And as it's still a draft, and not very widespread, I'm more concern on how our library consumer may interpret it. Even if both are technically valid, I would rather pick the more intuitive one.